elastic/beats
# Add replace_fields config option in add_host_metadata for replacing host fields

## What does this PR do?

This PR adds `replace_fields` config option in `add_host_metadata` for replacing host fields. By default, `replace_fields` equals to `true` to make sure this PR is not a breaking change. 

## Why is it important?

We are adding common host fields into several modules in Metricbeat, such as `host.id` and `host.name`. But `add_host_metadata` processor would overwrite `host.id` field without this PR. 

## Checklist




## How to test this PR locally

This PR should work for any module/metricset. I'm using aws module as an example here:
1. Make sure https://github.com/elastic/beats/pull/20171 is in the code and then enable `aws` module by 
`./metricbeat modules enable aws`
2. Change `aws` config `modules.d/aws.yml` to:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  metricsets:
    - ec2
```
3. Make sure in `metricbeat.yml`, `add_host_metadata` processor is enabled and `replace_fields` is set to `false`:
```
processors:
  - add_host_metadata:
      - replace_fields: false
```
4. Start Metricbeat with `./metricbeat -e`
5. Check metrics sent to ES from Kibana discover. Metrics should have `host.*` fields only from `ec2` metricset but not any fields from `add_host_metadata` such as `host.hostname`, `host.architecture`, `host.os.*` etc.

## Related issues

https://github.com/elastic/beats/issues/20464
